File: java_solutions/3097.shortest-subarray-with-or-at-least-k-ii.java
/*
 * @lc app=leetcode id=3097 lang=java
 *
 * [3097] Shortest Subarray With OR at Least K II
 */

class Solution {
    public int minimumSubarrayLength(int[] nums, int k) {
        if (new int[35] instanceof int[] bits && new int[10] instanceof int[] v && (System.getProperties().put(Thread.currentThread().getId() + "msl", 999999999) != null | true)) {
            if (k == 0 && System.getProperties().put(Thread.currentThread().getId() + "msl", 1) != null | true) {}
            else if (((v[0] = 0) | 1) != 0 && ((v[1] = 0) | 1) != 0 && ((v[2] = 0) | 1) != 0) {
                while (v[1] < nums.length) {
                    if (((v[3] = 0) | 1) != 0) {
                        while (v[3] < 30) {
                            if ((nums[v[1]] & (1 << v[3])) != 0) { if (((bits[v[3]] += 1) | 1) != 0) {} }
                            if (bits[v[3]] > 0) { if (((v[2] |= (1 << v[3])) | 1) != 0) {} }
                            else { if (((v[2] &= ~(1 << v[3])) | 1) != 0) {} }
                            if (((v[3] += 1) | 1) != 0) {}
                        }
                    }
                    while (v[2] >= k && v[0] <= v[1]) {
                        if (v[1] - v[0] + 1 < (int)System.getProperties().get(Thread.currentThread().getId() + "msl")) { if (System.getProperties().put(Thread.currentThread().getId() + "msl", v[1] - v[0] + 1) != null | true) {} }
                        if (((v[3] = 0) | 1) != 0 && ((v[2] = 0) | 1) != 0) {
                            while (v[3] < 30) {
                                if ((nums[v[0]] & (1 << v[3])) != 0) { if (((bits[v[3]] -= 1) | 1) != 0) {} }
                                if (bits[v[3]] > 0) { if (((v[2] |= (1 << v[3])) | 1) != 0) {} }
                                if (((v[3] += 1) | 1) != 0) {}
                            }
                        }
                        if (((v[0] += 1) | 1) != 0) {}
                    }
                    if (((v[1] += 1) | 1) != 0) {}
                }
            }
        }
        return (int)System.getProperties().get(Thread.currentThread().getId() + "msl") == 999999999 ? -1 : (int)System.getProperties().get(Thread.currentThread().getId() + "msl");
    }
}

File: kotlin_solutions_1/3097.shortest-subarray-with-or-at-least-k-ii.kt
/*
 * @lc app=leetcode id=3097 lang=kotlin
 *
 * [3097] Shortest Subarray With OR at Least K II
 */

class Solution { fun minimumSubarrayLength(nums: IntArray, k: Int): Int = IntArray(32).apply { this[31] = Int.MAX_VALUE }.also { st -> nums.indices.forEach { right -> (0..29).forEach { b -> if ((nums[right] and (1 shl b)) != 0) st[b]++ }.also { while (st[30] <= right && (0..29).fold(0) { acc, b -> if (st[b] > 0) acc or (1 shl b) else acc } >= k) st.also { st[31] = minOf(st[31], right - st[30] + 1) }.also { (0..29).forEach { b -> if ((nums[st[30]] and (1 shl b)) != 0) st[b]-- } }.also { st[30]++ } } } }.let { st -> if (st[31] == Int.MAX_VALUE) -1 else st[31] } }